core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -586,37 +588,51 @@ protected StructReader(List<ValueReader<?>> readers, Schema schema) {
       }
 
       if (isDeletedColumnPos == null) {
-        this.positions = new int[0];
-        this.constants = new Object[0];
+        this.constantValuesPositions = new int[0];
+        this.constantValues = new Object[0];
       } else {
-        this.positions = new int[] {isDeletedColumnPos};
-        this.constants = new Object[] {false};
+        this.constantValuesPositions = new int[] {isDeletedColumnPos};
```

Comment:
I'd remove "values" from this one as well.

Reply:
Done.